src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -158,54 +175,88 @@ private boolean executeHTTPServiceAndCheckStatusCode() throws IOException {
 
     if (accessToken != null) {
       Header authorizationHeader = new BasicHeader("Authorization",
-                                                   String.format("Bearer %s", accessToken.getTokenValue()));
+        String.format("Bearer %s", accessToken.getTokenValue()));
       headers.putAll(config.getHeadersMap(String.valueOf(authorizationHeader)));
     }
 
+    headers.put("Request-Method", config.getMethod().toUpperCase());
+    headers.put("Connect-Timeout", String.valueOf(config.getConnectTimeout()));
+    headers.put("Read-Timeout", String.valueOf(config.getReadTimeout()));
+    headers.put("Instance-Follow-Redirects", String.valueOf(config.getFollowRedirects()));
+    headers.put("charset", config.getCharset());
+
     try {
       URL url = new URL(configURL);
-      conn = (HttpURLConnection) url.openConnection();
-      if (conn instanceof HttpsURLConnection) {
-        //Disable SSLv3
+      HttpEntityEnclosingRequestBase request = new HttpClient.HttpRequest(URI.create(String.valueOf(url)),
+        config.getMethod());
+
+      if (!Strings.isNullOrEmpty(config.getProxyUrl())) {
+        URL proxyURL = new URL(config.getProxyUrl());
+        String proxyHost = proxyURL.getHost();
+        int proxyPort = proxyURL.getPort();
+        String proxyUser = config.getProxyUsername();
+        String proxyPassword = config.getProxyPassword();
+
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+
+        if (proxyUser != null && proxyPassword != null) {
+          credsProvider.setCredentials(
+            new AuthScope(proxyHost, proxyPort),
+            new UsernamePasswordCredentials(proxyUser, proxyPassword));
+        }
+
+        HttpHost proxy = new HttpHost(proxyHost, proxyPort);
+        httpClient = HttpClients.custom()
+          .setDefaultCredentialsProvider(credsProvider)
+          .setProxy(proxy)
+          .build();
+      }
```

Comment:
Refarctor this portion of the code into a separate method, getHttpClient internally which you can either create HttpClients.createDefault or HttpClients.custom based on the use case.

Reply:
Done